R/framework.R

```diff
@@ -66,12 +66,17 @@ trading_dates(start, end, calendar=holidayNYSE) %as% {
 trading_dates(start, obs, calendar=holidayNYSE) %::% a:numeric:Function:Date
 trading_dates(start, obs, calendar=holidayNYSE) %when% { obs>=0 } %as% {
   start <- as.Date(start)
-  # This is to get enough dates to account for holidays and weekends
-  shimmed <- ceiling(obs * 2)
-  dates <- timeSequence(from=start, length.out=shimmed)
-  dates <- as.Date(dates[isBizday(dates, holidays=calendar(unique(year(dates))))])
-  dates <- dates[dates >= start]
-  dates <- dates[1:obs]
+  # This loop is to make sure we can always get enough dates to account for holidays and weekends
+  shimmed <- obs
+  repeat{
```

Comment:
One issue is that every iteration is re-generating values, which is inefficient. It is better to use a functional programming approach that more clearly describes what happens in each iteration.
e.g.
```
step.size <- 100
max.its <- ceiling(2 * obs / step.size)
callCC(function(ext) fold(1:max.its, function(i, dates) {
  if (length(dates) >= obs) return(ext(dates[1:obs]))
  ... # old code here
}, NULL))
```